app/src/main/java/endava/codebase/android/movieapp/ui/component/MovieCard.kt

```diff
@@ -0,0 +1,75 @@
+package endava.codebase.android.movieapp.ui.component
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.paint
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.modifier.modifierLocalConsumer
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.rememberAsyncImagePainter
+import endava.codebase.android.movieapp.R
+
+data class MovieCardViewState(
+    val id: Int,
+    val title: String,
+    val overview: String,
+    val imageUrl: String?,
+    val isFavorite: Boolean,
```

Comment:
You only need `imageUrl` and `isFavorite` in the `MovieCard` component, the rest can be removed.